### PR TITLE
Sdk profiler

### PR DESCRIPTION
**Pull request by irexyc**
## Motivation

Add sdk profiler.

### CODE REVIEW

**Comment by lvhan028** [comment]
May merge master to resolve circleci error

File: csrc/mmdeploy/apis/c/mmdeploy/common.h [hunk 1]
```diff
@@ -123,6 +126,20 @@ MMDEPLOY_API int mmdeploy_device_create(const char* device_name, int device_id,
  */
 MMDEPLOY_API void mmdeploy_device_destroy(mmdeploy_device_t device);
 
+/**
+ * Create profiler
```
Comment (lvhan028) [inline csrc/mmdeploy/apis/c/mmdeploy/common.h:130]: Please clarify each parameter and the return value

File: csrc/mmdeploy/graph/CMakeLists.txt [hunk 2]
```diff
@@ -3,10 +3,14 @@
...
 set(SRCS
+        profiler.cpp
         task.cpp
         static_router.cpp
         inference.cpp
         pipeline.cpp
         cond.cpp)
 mmdeploy_add_module(${PROJECT_NAME} LIBRARY "${SRCS}")
+target_include_directories(${PROJECT_NAME}
+        PUBLIC
```
Comment (lvhan028) [inline csrc/mmdeploy/graph/CMakeLists.txt:14]: Can we use 'PRIVATE'?

File: csrc/mmdeploy/apis/c/mmdeploy/common.cpp [hunk 3]
```diff
@@ -72,6 +73,17 @@ int mmdeploy_device_create(const char* device_name, int device_id, mmdeploy_devi
...
 void mmdeploy_device_destroy(mmdeploy_device_t device) { delete (Device*)device; }
 
+int mmdeploy_profiler_create(const char* path, mmdeploy_profiler_t* profiler) {
+  *profiler = (mmdeploy_profiler_t) new profiler::Profiler(path);
+  return MMDEPLOY_SUCCESS;
+}
+
+void mmdeploy_profiler_destroy(mmdeploy_profiler_t profiler) {
+  auto p = (profiler::Profiler*)profiler;
+  p->Release();
```
Comment (lvhan028) [inline csrc/mmdeploy/apis/c/mmdeploy/common.cpp:83]: `p` should be validated before accessing its member function

File: csrc/mmdeploy/graph/profiler.cpp [hunk 4]
```diff
@@ -0,0 +1,82 @@
+#include "mmdeploy/graph/profiler.h"
```
Comment (lzhangzz) [inline csrc/mmdeploy/graph/profiler.cpp:1]: Profiler should be part of mmdeploy_core

File: csrc/mmdeploy/graph/profiler.h [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+struct MMDEPLOY_API Profiler {
+  explicit Profiler(std::string_view path);
+  ScopeSptr CreateScope(std::string_view name);
+  Event* AddEvent(Event e);
+  ScopeSptr scope() const noexcept { return root_; }
+  void Release();
+
+  std::string path_;
+  std::vector<ScopeSptr> nodes_;
```
Comment (lzhangzz) [inline csrc/mmdeploy/graph/profiler.h:77]: Profiler should be the unique owner of all scopes, consider using `std::deque<Scope>`, and refer to its elements by `Scope*`.

**Review by lzhangzz — APPROVED** [review]
LGTM

File: csrc/mmdeploy/apis/cxx/mmdeploy/common.hpp [hunk 6]
```diff
@@ -76,6 +76,24 @@ class Device {
   std::shared_ptr<mmdeploy_device> device_;
 };
 
+class Profiler {
+ public:
+  explicit Profiler(std::string path) : path_(path) {
```
Comment (lvhan028) [inline csrc/mmdeploy/apis/cxx/mmdeploy/common.hpp:81]: `const std::string& path`

File: csrc/mmdeploy/core/profiler.cpp [hunk 7]
```diff
@@ -0,0 +1,81 @@
+#include "mmdeploy/core/profiler.h"
```
Comment (lvhan028) [inline csrc/mmdeploy/core/profiler.cpp:1]: May add the license part

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**